File: data/sample_data.js
// todo_manager.js
const fs = require("fs");
const readline = require("readline");

const TODO_FILE = "todo.txt";

function loadTasks() {
    if (fs.existsSync(TODO_FILE)) {
        const data = fs.readFileSync(TODO_FILE, "utf8");
        return data.split("\n").filter(task => task.trim() !== "");
    }
    return [];
}

function saveTasks(tasks) {
    fs.writeFileSync(TODO_FILE, tasks.join("\n"), "utf8");
}

function addTask(tasks) {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
    });
    rl.question("Enter a new task: ", (task) => {
        tasks.push(task);
        saveTasks(tasks);
        console.log(`Task added: ${task}`);
        rl.close();
        main();
    });
}

function viewTasks(tasks) {
    if (tasks.length === 0) {
        console.log("No tasks found.");
    } else {
        console.log("Your tasks:");
        tasks.forEach((task, index) => {
            console.log(`${index + 1}. ${task}`);
        });
    }
}

function deleteTask(tasks) {
    viewTasks(tasks);
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
    });
    rl.question("Enter the task number to delete: ", (taskNum) => {
        const index = parseInt(taskNum) - 1;
        if (index >= 0 && index < tasks.length) {
            const removedTask = tasks.splice(index, 1);
            saveTasks(tasks);
            console.log(`Task deleted: ${removedTask}`);
        } else {
            console.log("Invalid task number.");
        }
        rl.close();
        main();
    });
}

function main() {
    const tasks = loadTasks();
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
    });
    console.log("\nTo-Do List Manager");
    console.log("1. View Tasks");
    console.log("2. Add Task");
    console.log("3. Delete Task");
    console.log("4. Exit");
    rl.question("Choose an option: ", (choice) => {
        switch (choice) {
            case "1":
                viewTasks(tasks);
                rl.close();
                main();
                break;
            case "2":
                rl.close();
                addTask(tasks);
                break;
            case "3":
                rl.close();
                deleteTask(tasks);
                break;
            case "4":
                console.log("Goodbye!");
                rl.close();
                process.exit();
                break;
            default:
                console.log("Invalid choice. Please try again.");
                rl.close();
                main();
                break;
        }
    });
}

main();

File: data/sample_data.py
# todo_manager.py
import os

TODO_FILE = "todo.txt"

def load_tasks():
    if os.path.exists(TODO_FILE):
        with open(TODO_FILE, "r") as file:
            return file.read().splitlines()
    return []

def save_tasks(tasks):
    with open(TODO_FILE, "w") as file:
        for task in tasks:
            file.write(task + "\n")

def add_task(tasks):
    task = input("Enter a new task: ")
    tasks.append(task)
    save_tasks(tasks)
    print(f"Task added: {task}")

def view_tasks(tasks):
    if not tasks:
        print("No tasks found.")
    else:
        print("Your tasks:")
        for i, task in enumerate(tasks, 1):
            print(f"{i}. {task}")

def delete_task(tasks):
    view_tasks(tasks)
    try:
        task_num = int(input("Enter the task number to delete: "))
        if 1 <= task_num <= len(tasks):
            removed_task = tasks.pop(task_num - 1)
            save_tasks(tasks)
            print(f"Task deleted: {removed_task}")
        else:
            print("Invalid task number.")
    except ValueError:
        print("Please enter a valid number.")

def main():
    tasks = load_tasks()
    while True:
        print("\nTo-Do List Manager")
        print("1. View Tasks")
        print("2. Add Task")
        print("3. Delete Task")
        print("4. Exit")
        choice = input("Choose an option: ")
        if choice == "1":
            view_tasks(tasks)
        elif choice == "2":
            add_task(tasks)
        elif choice == "3":
            delete_task(tasks)
        elif choice == "4":
            print("Goodbye!")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()